pkg/ring/util.go

```diff
@@ -34,6 +34,11 @@ func GenerateTokens(numTokens int, takenTokens []uint32) []uint32 {
 		i++
 	}
 
+	// Ensure returned tokens are sorted.
+	sort.Slice(tokens, func(i, j int) bool {
```

Comment:
In the ring lifecycler it's already guaranteed all tokens are sorted. I've introduced sorting here as well to have same guarantee in tests too.